### PR TITLE
Users list

### PR DESCRIPTION
**Pull request by IamBeaver**
Have created users list and user list item components. Created store for users list that contains array of users. Injected it. Created user model. Fixed bugs in Users and RouteBuilder services.

### CODE REVIEW

File: src/ioc/ioc.ts [hunk 1]
```diff
@@ -2,9 +2,12 @@ import { Container } from "inversify";
...
+import UsersListStore from "../stores/UsersListStore";
 import ownTypes from "./ownTypes";
 
 export const container = new Container();
 container.bind<HttpService>(ownTypes.httpService).to(DefaultHttpService).inSingletonScope();
 container.bind<UsersService>(ownTypes.usersService).to(DefaultUsersService).inSingletonScope();
-container.bind<RouteBuilderService>(ownTypes.routeBuilderService).to(DefaultRouteBuilderService).inTransientScope();
\ No newline at end of file
+container.bind<RouteBuilderService>(ownTypes.routeBuilderService).to(DefaultRouteBuilderService).inTransientScope();
+
```
Comment (MarikNez) [inline src/ioc/ioc.ts]: styleCode- delete empty enter

File: src/components/UserListItem/UserListItem.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Avatar, IconButton, ListItem, ListItemAvatar, ListItemText } from "@mui/material";
+import { observer } from "mobx-react";
+import { Person, Delete } from "@mui/icons-material";
+
+interface Props {
```
Comment (anniemt6) [inline src/components/UserListItem/UserListItem.tsx]: `IUserListItemProps` instead `Props`.

File: src/components/UserListItem/UserListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    id: number;
+    firstName: string;
+    lastName: string;
+    age: number;
+}
+
+const UserListItem = observer(({id, firstName, lastName, age}: Props) => {
```
Comment (MarikNez) [inline src/components/UserListItem/UserListItem.tsx]: Naming - we use multiple Users in List. And you already have UsersList in code belowe

File: src/components/UsersList/UsersList.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { List } from "@mui/material";
+import { observer } from "mobx-react";
+import { useEffect } from "react";
+import { useInjection } from "../../ioc/ioc.react";
+import ownTypes from "../../ioc/ownTypes";
+import UserListStore from "../../stores/UsersListStore";
+import UserListItem from "../UserListItem";
+
+const UsersList = observer(() => {
+    const store = useInjection<UserListStore>(ownTypes.usersListStore);
```
Comment (MarikNez) [inline src/components/UsersList/UsersList.tsx]: Does it actually work? There is no UserListStore in UsersListStore

File: src/components/UserListItem/UserListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                <IconButton edge='end'>
+                    <Delete/>
+                </IconButton>
+            }
+        >
+                <ListItemAvatar>
+                    <Avatar>
+                        <Person/>
+                    </Avatar>
+                </ListItemAvatar>
```
Comment (anniemt6) [inline src/components/UserListItem/UserListItem.tsx]: Maybe it`s better to create the component for that.

File: src/models/User.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export default interface User {
```
Comment (anniemt6) [inline src/models/User.ts:1]: Name interfaces like that: `IUser`.

File: src/stores/UsersListStore.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        makeAutoObservable(this);
+    }
+
+    public init = async(): Promise<void> => {
+        try {
+           this.users = await this.usersService.getAllUsers();
+        } catch (e) {
+            if (e instanceof Error) {
+                console.log(e.message);
+            }
```
Comment (anniemt6) [inline src/stores/UsersListStore.ts]: And you will check in every catch construction instance of Error? Maybe it's better to create a method od just remove the check.

Comment (anniemt6) [reply src/stores/UsersListStore.ts]: In addition it's not a best practice to write logs in the code. You should think how to handle errors using another ideas.

File: src/models/User.ts (hunk 6 above)
Comment (IamBeaver) [reply src/models/User.ts:1]: convention says that i shouldn't use I in interface naming.

**Pull request merged**